File: Python/55. Jump Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        end=0
        idx=0
        n=len(nums)
        for i in range(0,n):
            idx=max(idx,i+nums[i])
            if i==end:
                end=idx
            
        
        if end>=n-1:
            return True
        return False

File: Java/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int end=0;
        int idx=0;
        int n=nums.length;
        for(int i=0;i<n;i++){
            idx=Math.max(idx,i+nums[i]);
            if(i==end){
                end=idx;
            }
        }
        if(end>=n-1){return true;}
        return false;
    }
}
